### PR TITLE
Sanitation checks

### LINKED ISSUE #6: Sanitation checks
**Is your feature request related to a problem?**
Move features on submission sanitization, from further down the pipeline to the current dequeuer component:

* discard submissions from snapshotter identities absuing network resources with their submissions
* verify the pair contract index being submitted against the time slot of the day and the slot ID 

These should not be left to the batcher to consider or discard, and instead the dequeuer should discard as early as possible.

**Describe the solution you'd like**

Extend the `verifyAndStoreSubmission` to integrate the above features.

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by chetna-mittal**
Fixes #6 

### Checklist
- [x] My branch is up-to-date with upstream/main branch.
- [x] Everything works and tested for major version of Python/NodeJS/Go and above.
- [x] I ran pre-commit checks against my changes.
- [x] I've written tests against my changes and all the current present tests are passing.

### Current behaviour

Currently, `verifyAndStoreSubmission` function does not have sanitation checks to discard submissions from snapshotters abusing network resources with their submissions and for verifying the pair contract index. 

### New expected behaviour

- Flagged snapshotters are discarded and 
- Pair contract index is now verified

### CODE REVIEW

File: config/settings.go [hunk 1]
```diff
@@ -36,6 +41,12 @@ func LoadConfig() {
 		log.Fatalf("Failed to parse CHAIN_ID environment variable: %v", err)
 	}
 
+	url := getEnv("SETTINGS_URL", "")
```
[the remark this answers is not in the document]
Comment (chetna-mittal) [reply config/settings.go]: Accommodated: 325e8c030ff741bcaf1419a827539d38b670932d

File: config/settings.go [hunk 2]
```diff
@@ -36,6 +41,12 @@ func LoadConfig() {
...
+	initialPairs, err := fetchInitialPairs(url)
```
[the remark this answers is not in the document]
Comment (chetna-mittal) [reply config/settings.go]: Accommodated: 325e8c030ff741bcaf1419a827539d38b670932d

File: pkgs/prost/contract.go [hunk 3]
```diff
@@ -47,3 +53,62 @@ func MustQuery[K any](ctx context.Context, call func() (val K, err error)) (K, e
 	}
 	return val, err
 }
+
+func getExpirationTime(epochID int64) time.Time {
+	// DAY_SIZE in microseconds
+	daySize := 864000000 * time.Microsecond
```
[the remark this answers is not in the document]
Comment (chetna-mittal) [reply pkgs/prost/contract.go]: Accommodated: 1b7526ec83d9cf1f5c80788b2bbd8c2e5f806d37

**Comment by anomit** [comment]
The PR is blocked on the resolution of this issue https://github.com/PowerLoom/snapshotter-lite-v2/issues/65 which will standardize the hashing algorithm used to generate the pair contract indexes that a slot ID is responsible for a specific time slot within the day.

**Comment by anomit** [comment]
Closing this PR as the commits from the contributor's fork has been incorporated into the `dockerify` branch, along with a feature flag to activate the check on pair contract indexes only when necessary so that .

Said commits can be found [here](https://github.com/PowerLoom/sequencer-dequeuer/compare/main...dockerify) which will be included in a future PR from a fork of `dockerify` to `main` that addresses issue #8

**Pull request closed without merging**